ocaml/xapi-storage-script/main.ml

```diff
@@ -1549,13 +1549,20 @@ let bind ~volume_script_dir =
   in
   S.VDI.list_changed_blocks vdi_list_changed_blocks_impl ;
   let vdi_data_destroy_impl dbg sr vdi =
-    wrap
-    @@
-    let* sr = Attached_SRs.find sr in
-    let vdi = Storage_interface.Vdi.string_of vdi in
-    return_volume_rpc (fun () ->
-        Volume_client.data_destroy volume_rpc dbg sr vdi
-    )
+    Attached_SRs.find sr
+    >>>= (fun sr ->
+           let vdi =
+             Storage_interface.Vdi.string_of vdi
+           in
+           return_volume_rpc (fun () ->
+               Volume_client.data_destroy volume_rpc dbg sr vdi
+           )
+           >>>= fun response ->
+           set ~dbg ~sr ~vdi ~key:_vdi_type_key ~value:"cbt_metadata"
+           >>>= fun () ->
+           Deferred.Result.return response
+         )
+    |> wrap
```

Comment:
To reduce churn, the new code can continue using the let* idiom:

```suggestion
    wrap
    @@
    let* sr = Attached_SRs.find sr in
    let vdi = Storage_interface.Vdi.string_of vdi in
    let* response =
      return_volume_rpc (fun () ->
          Volume_client.data_destroy volume_rpc dbg sr vdi
      )
    in
    let* () = set ~dbg ~sr ~vdi ~key:_vdi_type_key ~value:"cbt_metadata" in
    Deferred.Result.return response
```

Reply:
Done